app/src/test/java/com/github/multimatum_team/multimatum/QRGeneratorActivityTest.kt

```diff
@@ -14,27 +18,35 @@ import androidx.test.espresso.matcher.BoundedMatcher
 import androidx.test.espresso.matcher.ViewMatchers
 import androidx.test.ext.junit.rules.ActivityScenarioRule
 import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.github.multimatum_team.multimatum.service.ClockService
+import com.github.multimatum_team.multimatum.util.QRGeneratorUtilityTest
+import com.google.android.apps.common.testing.accessibility.framework.utils.contrast.Image
 import com.google.zxing.*
 import com.google.zxing.common.HybridBinarizer
 import com.google.zxing.qrcode.QRCodeReader
+import dagger.hilt.android.testing.HiltAndroidRule
+import dagger.hilt.android.testing.HiltAndroidTest
+import dagger.hilt.android.testing.UninstallModules
 import org.hamcrest.CoreMatchers.equalTo
 import org.hamcrest.Description
 import org.hamcrest.Matcher
 import org.hamcrest.MatcherAssert.assertThat
 import org.hamcrest.Matchers
+import org.junit.Before
 import org.junit.Rule
 import org.junit.Test
 import org.junit.runner.RunWith
 
 import org.robolectric.shadows.ShadowToast
+import javax.inject.Inject
 
 /**
  * Tests for the QRGenerator class.
  */
 @RunWith(AndroidJUnit4::class)
-class QRGeneratorTest {
+class QRGeneratorActivityTest: QRGeneratorUtilityTest {
     @get:Rule
-    val activityRule = ActivityScenarioRule(QRGenerator::class.java)
+    val activityRule = ActivityScenarioRule(QRGeneratorActivity::class.java)
```

Comment:
As `qRDisplayTest()` need to go to QRGeneratorActivity, I'm not sure that this is a good idea to put this as it will launch the class once without any intent and I don't know how the app react to this. 
You could in a @before put 

 `val intent = Intent(ApplicationProvider.getApplicationContext(),QRGeneratorActivity::class.java).putExtra(EXTRA_ID, "1")
ActivityScenario.launch<QRGeneratorActivity>(intent)`
        
to launch all tests correctly but I'm not really sure about that. If you think that what you have done is better, do not change anything

Reply:
what you are proposing is basically the same thing. As in my tests a sent a new intent it "update" the activity so it works too.